### ISSUE TITLE
Unable to build ParallelInfo in Singularity

### ISSUE THREAD

**Issue by fmahebert** [sxs-collaboration/spectre#969]
### Expected behavior:

Compiling SpECTRE in Singularity container succeeds.

### Current behavior:

SpECTRE on today's `develop`
`spectre.img` built by Singularity according to installation instructions
New clean build dir for CMake and make invocations

Configuring CMake for `clang++`, get errors:
```
[ 27%] Linking CXX executable ../../../bin/ParallelInfo
In file included from ParallelInfo_InfoFromBuild.cpp:4:
/home/francois/code/spectre-work/spectre/src/Informer/InfoFromBuild.hpp:49:30: error: expected function body after function declarator
std::string unit_test_path() noexcept;
                             ^
ParallelInfo_InfoFromBuild.cpp:28:35: error: expected expression
  static const std::string info = [] {
                                  ^
ParallelInfo_InfoFromBuild.cpp:43:30: error: expected function body after function declarator
std::string unit_test_path() noexcept {
                             ^
3 errors generated.
```

Configuring CMake for `g++`, get errors:
```
[ 27%] Linking CXX executable ../../../bin/ParallelInfo
/usr/bin/ld.gold: error: /home/francois/code/charm/multicore-linux64/lib/libmoduleDistributedLB.a(DistributedLB.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /home/francois/code/charm/multicore-linux64/lib/libmoduleDistributedLB.a(DistributedLB.o): requires dynamic R_X86_64_PC32 reloc against 'memset' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /home/francois/code/charm/multicore-linux64/lib/libmoduleDistributedLB.a(DistributedLB.o): requires dynamic R_X86_64_PC32 reloc against 'malloc' which may overflow at runtime; recompile with -fPIC
/usr/bin/ld.gold: error: /home/francois/code/charm/multicore-linux64/lib/libmoduleDistributedLB.a(DistributedLB.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
....
[ad infinitum]
```

### Environment:

The `g++` tree `$SPECTRE_BUILD_DIR/LibraryVersions.txt` is
```
CMake Version: 3.9.1
SpECTRE Version: 0.0.0
SpECTRE branch: develop
SpECTRE hash: 3715147f4a2fd67b33f2797e556086d9d0cb7b78
Hostname: noisette
Host system: Linux-4.17.12-100.fc27.x86_64
Host system version: 4.17.12-100.fc27.x86_64
Host system processor: x86_64
CMAKE_CXX_FLAGS:
CMAKE_CXX_LINK_FLAGS:
CMAKE_CXX_FLAGS_DEBUG: -g
CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_C_FLAGS:
CMAKE_C_LINK_FLAGS:
CMAKE_C_FLAGS_DEBUG: -g
CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_Fortran_FLAGS:
CMAKE_Fortran_LINK_FLAGS:
CMAKE_Fortran_FLAGS_DEBUG: -g
CMAKE_Fortran_FLAGS_RELEASE: -O3 -DNDEBUG -O3
CMAKE_EXE_LINKER_FLAGS:
CMAKE_BUILD_TYPE: Debug
CMAKE_CXX_COMPILER: /usr/bin/g++
CMAKE_CXX_COMPILER_VERSION: 7.2.0
CMAKE_C_COMPILER: /usr/bin/gcc
CMAKE_C_COMPILER_VERSION: 7.2.0
CMAKE_Fortran_COMPILER: /usr/bin/gfortran
CMAKE_Fortran_COMPILER_VERSION: 7.2.0
Python Version: 2.7.14
Charm Version:  6.8.0
BLAS_LIBRARIES:  /usr/lib/x86_64-linux-gnu/libopenblas.so
Blaze Version:  3.2
Boost Version:  1.62.0
Catch Version:  2.1.0
Google Benchmark Found
GSL Version:  2.4
HDF5 Version:  1.10.0.1
jemalloc Version:  ..
LIBXSMM Version:  1.8.1
LAPACK_LIBRARIES:  /usr/lib/x86_64-linux-gnu/libopenblas.so;/usr/lib/x86_64-linux-gnu/libopenblas.so
NumPy Version:  1.14.2
```

**Comment by nilsdeppe** [comment]
Can you add parens to the lambda expression and check if that works? If so, can you submit a PR?

**Comment by fmahebert** [comment]
Assuming you mean `[] {` -> `[]() {`, then this does not work...

Clang:
```
...
ParallelInfo_InfoFromBuild.cpp:28:35: error: expected expression
  static const std::string info = []() {
                                  ^
...
```

GCC: same errors as pasted above

**Comment by fmahebert** [comment]
TL;DR: issue is invalid.

I had confused the several different charm++ builds on my system and used one from outside the container :(. With the correct charm++ from inside the container, SpECTRE builds as expected.
